src/Microsoft.TemplateEngine.Edge/Installers/Folder/FolderInstaller.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Abstractions.GlobalSettings;
+using Microsoft.TemplateEngine.Abstractions.Installer;
+using Microsoft.TemplateEngine.Abstractions.TemplatePackages;
+
+namespace Microsoft.TemplateEngine.Edge.Installers.Folder
+{
+    internal class FolderInstaller : IInstaller
+    {
+        private readonly IEngineEnvironmentSettings _settings;
+
+        public FolderInstaller(IEngineEnvironmentSettings settings, IInstallerFactory factory, IManagedTemplatePackagesProvider provider)
+        {
+            Name = factory.Name;
+            FactoryId = factory.Id;
+            _settings = settings;
+            Provider = provider;
+        }
+
+        public Guid FactoryId { get; }
+        public string Name { get; }
+        public IManagedTemplatePackagesProvider Provider { get; }
+
+        public Task<bool> CanInstallAsync(InstallRequest installationRequest, CancellationToken cancellationToken)
+        {
+            _ = installationRequest ?? throw new ArgumentNullException(nameof(installationRequest));
+
+            return Task.FromResult(Directory.Exists(installationRequest.Identifier));
+        }
+
+        public IManagedTemplatePackage Deserialize(IManagedTemplatePackagesProvider provider, TemplatePackageData data)
```

Comment:
That's a good catch thanks. At some point we got side tracked from original idea and put Provider property to Installer, though Installer instance can be used by several providers. 
I have addressed it here: https://github.com/dotnet/templating/pull/2971/commits/c924c50250bcefb82410fb9af99f7f517b626b50

Provider is now passed to `IManagedTemplatePackage` from `Deserialize`, so it can be accessed through `IManagedTemplatePackage` interface.
Instead of storing provider in Installer, now all methods have parameter for passing the provider that called the method.
Now, in `NuGetInstaller` and `FolderInstaller` provider is not used in all methods, but having it in interface is beneficial for those who might want to implement their `IInstaller`s and used it in the methods.